src/components/HeaderComponent/Header.js

```diff
@@ -1,38 +1,70 @@
-import "../HeaderComponent/Header.scss"
+import React, { Component } from 'react';
+import "../HeaderComponent/Header.scss";
+import preview from "../../assets/images/preview.svg";
+import Basket from "./Basket/Basket";
+import HeaderButton from "./HeaderButton/HeaderButton";
+import { btnName } from "../../constants/constants";
 
-import preview from "../../assets/images/preview.svg"
-import purchases from "../../assets/images/purchases.svg"
+class Header extends Component {
 
-import HeaderButton from "./HeaderButton/HeaderButton";
+    constructor(props) {
+        super(props);
+        this.state = {
```

Comment:
It looks like the state for this component is an extra thing. It requires you to use componentDidUpdate and some calculations inside of the render method.